### PR TITLE
Break: Add gradle 6.0-20190904072820+0000 compatibility. Compile with -Werror and -Xlint:deprecation

### PR DESCRIPTION
**Pull request by robert3005**
## Before this PR
Compilation would fail if baseline was used with gradle 6 snapshots due to deprecations being removed

## After this PR
==COMMIT_MSG==
Add gradle 6.0-20190904072820+0000 compatibility. This raises minimum required version of gradle for plugins from this repo to 5.0. Additionally add '-Werror' and '-Xlint:deprecation' so we can detect compatibility issues early
==COMMIT_MSG==

## Possible downsides?
This pr is naive and just requires min gradle version to be 5 instead of 4.10

### CODE REVIEW

File: changelog/@unreleased/pr-791.v2.yml
Comment (ferozco) [inline changelog/@unreleased/pr-791.v2.yml]: Could you describe the change in a way that would be helpful for consumers of the plugin? something like: `gradle-baseline should now be compatible with Gradle  6.0. gradle-baseline now requires at least Gradle 5.0 and is no longer compatible with 4.X.`

Comment (robert3005) [reply changelog/@unreleased/pr-791.v2.yml]: I am not sure we are compatible with gradle 6 though since it's not released. I could say it's compatible with gradle snapshot that I tested with.

Comment (robert3005) [reply changelog/@unreleased/pr-791.v2.yml]: Also if I was just making it compatible with gradle 6 I wouldn't be adding the compiler flags

File: changelog/@unreleased/pr-791.v2.yml
Comment (schlosna) [inline changelog/@unreleased/pr-791.v2.yml]: we should flag [this as a break](https://github.com/palantir/gradle-baseline/pull/493#issuecomment-451897295) so any folks lagging on gradle < 5.x upgrade

Comment (robert3005) [reply changelog/@unreleased/pr-791.v2.yml]: As I mentioned in the pr description it's possible to not make it a break so I leave it to the maintainers to decide what they prefer

File: gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineIntegrationTest.groovy
Comment (iamdanfox) [inline gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineIntegrationTest.groovy]: could we add `6.0-20190904072820+0000` here?

File: changelog/@unreleased/pr-791.v2.yml
Comment (iamdanfox) [reply changelog/@unreleased/pr-791.v2.yml]: Yeah I'd just phrase this as a break and declare compatibility with `6.0-20190904072820+0000`.  Seems like we've got rid of most gradle4 users in the foundry org:

Screenshot 2019-09-04 at 15 39 47

Comment (iamdanfox) [reply changelog/@unreleased/pr-791.v2.yml]: Mmmm looks like there are more stragglers outside Foundry, but I think this is probably OK:

image

File: gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineIntegrationTest.groovy
Comment (robert3005) [reply gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineIntegrationTest.groovy]: sure

**Review by iamdanfox — APPROVED** [review]
Ok this will cause gradle-baseline 2.0.  Anything else we want to squeeze in before we cut this?

**Pull request merged**
